rectools/models/popular.py

```diff
@@ -41,17 +43,83 @@ class Popularity(Enum):
     SUM_WEIGHT = "sum_weight"
 
 
+def _serialize_timedelta(td: tp.Optional[tp.Union[None, dict, timedelta]]) -> tp.Optional[timedelta]:
+    if isinstance(td, dict):
+        return timedelta(
+            days=td.get("days", 0),
+            seconds=td.get("seconds", 0),
+            microseconds=td.get("microseconds", 0),
+            milliseconds=td.get("milliseconds", 0),
+            minutes=td.get("minutes", 0),
+            hours=td.get("hours", 0),
+            weeks=td.get("weeks", 0),
+        )
+    return td
+
+
+def _deserialize_timedelta(td: tp.Optional[timedelta]) -> tp.Optional[dict]:
+    if td is None:
+        return td
+    return {"days": td.days, "seconds": td.seconds, "microseconds": td.microseconds}
+
+
+TimeDelta = tpe.Annotated[
+    tp.Union[None, timedelta, dict],
+    PlainValidator(func=_serialize_timedelta),
+    PlainSerializer(func=_deserialize_timedelta),
+]
+
+
 class PopularModelConfig(ModelConfig):
     """Config for `PopularModel`."""
 
     popularity: Popularity = Popularity.N_USERS
-    period: tp.Optional[timedelta] = None
-    begin_from: tp.Optional[datetime] = None
+    period: TimeDelta = None
+    begin_from: tp.Optional[tp.Union[datetime, str]] = None
     add_cold: bool = False
     inverse: bool = False
 
 
-class PopularModel(FixedColdRecoModelMixin, ModelBase):
+class PopularModelMixin:
+    """Mixin for models based on popularity."""
+
+    def _validate_popular_model_attributes(
+        self,
+        popularity: tp.Literal["n_users", "n_interactions", "mean_weight", "sum_weight"],
+        period: TimeDelta,
+        begin_from: tp.Optional[tp.Union[datetime, str]],
+    ) -> None:
+        try:
+            self.popularity = Popularity(popularity)
+        except ValueError:
+            possible_values = {item.value for item in Popularity.__members__.values()}
+            raise ValueError(f"`popularity` must be one of the {possible_values}. Got {popularity}.")
+        if period is not None and begin_from is not None:
+            raise ValueError("Only one of `period` and `begin_from` can be set")
+
+    def _filter_interactions(
```

Comment:
You don't use `self` inside, hence you can make it a classmethod. As well as other methods here

Reply:
Done.